tdms/src/interpolate.cpp

```diff
@@ -1393,3 +1272,177 @@ void interpolateTimeDomainFieldCentralE_TM(  double ***Exy, double ***Exz, doubl
   *Ey = 0.;
   *Ez = Ezy[k][j][i]+Ezx[k][j][i];
 }
+
+/**
+ * @brief Determines the appropriate interpolation scheme to use
+ * 
+ * @param[in] cells_in_direction The number of Yee cells in the interpolation direction of interest
+ * @param[in] cell_id The current ID (in this dimension) of the Yee cell
+ * @return int Equal to BAND_LIMITED, INTERP1, INTERP2, INTERP3, indicating the appropriate interpolation scheme
+ */
+int determineInterpScheme(int cells_in_direction, int cell_id) {
+  // interpolation is impossible if the total number of cells in this direction is <4
+  if (cells_in_direction <= 4) { throw out_of_range("Error: computational domain has fewer than 4 cells in at least 1 dimension, cubic and band-limited interpolation impossible.\n"); }
```

Comment:
from a code style point of view I'd prefer if these `throw` s were on the next line. but we are definitely going to need to clang-format everything anyway